packages/core/src/consumed-thing.ts

```diff
@@ -555,7 +557,33 @@ export default class ConsumedThing extends TD.Thing implements IConsumedThing {
         form = this.handleUriVariables(tp, form, options);
 
         const content = await client.readResource(form);
-        return new InteractionOutput(content, form, tp);
+        try {
+            return this.handleInteractionOutput(content, form, tp);
+        } catch (e) {
+            const error = e instanceof Error ? e : new Error(JSON.stringify(e));
+            throw new Error(`Error while processing property for ${tp.title}. ${error.message}`);
+        }
```

Comment:
I am wondering if we could maybe also simplify these try-catch-blocks, as the `handleInteractionOutput` method itself is probably the only place where an error could originate from, right? But this would only be an optimization step.

Reply:
I was thinking about the same. In the first commits the `try catch` block was inside `handleInteractionOutput()`. 

The downside was/is that either the error message is not very clear (not stating for which interaction name & type it originates from) OR we need to pass additional arguments like interaction-type (property, action, event) and interaction-name is used to `handleInteractionOutput()` purely for logging.
Personally I didn't like that much and hence decided to do the error handling in each interaction even though I know it duplicates code.

Anyhow, I am open to either way...

Reply:
> The downside was/is that either the error message is not very clear (not stating for which interaction name & type it originates from) OR we need to pass additional arguments like interaction-type (property, action, event) and interaction-name is used to handleInteractionOutput() purely for logging.
> Personally I didn't like that much and hence decided to do the error handling in each interaction even though I know it duplicates code.

Yeah, I agree, passing these parameters only for logging is definitely not that ideal. So I think we can keep it as is :) Maybe one potential solution for the future could be introducing a common interface for `ConsumedThingProperty`, `ConsumedThingAction`, `ConsumedThinEvent` and then passing the respective interaction affordance to the handler function which could then obtain the necessary information for logging.